plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AmazonS3Fixture.java

```diff
@@ -70,17 +72,24 @@ private AmazonS3Fixture(final String workingDir, final String permanentBucketNam
 
     @Override
     protected Response handle(final Request request) throws IOException {
-        final RequestHandler handler = handlers.retrieve(request.getMethod() + " " + request.getPath(), request.getParameters());
+        final String nonAuthorizedPath = "* " + request.getMethod() + " " + request.getPath();
+        final RequestHandler nonAuthorizedHandler = handlers.retrieve(nonAuthorizedPath, request.getParameters());
+        if (nonAuthorizedHandler != null) {
+            return nonAuthorizedHandler.handle(request);
+        }
+
+        final String authorizedPath = "A " + request.getMethod() + " " + request.getPath();
+        final RequestHandler handler = handlers.retrieve(authorizedPath, request.getParameters());
         if (handler != null) {
             final String authorization = request.getHeader("Authorization");
             final String permittedBucket;
-            if (authorization.contains("s3_integration_test_permanent_access_key")) {
+            if (authorization != null && authorization.contains("s3_integration_test_permanent_access_key")) {
```

Comment:
Could we handle the `authorisation == null` case first to avoid these null checks proliferating? It's also helpful to distinguish no authorisation header vs a bad one, which this would achieve.